iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,214 @@
+        {
+            ctx.fibonacciLast = ctx.fibonacciCurrent;
+            ctx.fibonacciCurrent = response->sum;
+            std::cout << APP_NAME << " Got Response : " << ctx.fibonacciCurrent << std::endl;
+        }
+        else
+        {
+            std::cout << "Got Response with outdated sequence ID! Expected = "
+                      << ctx.expectedResponseSequenceId << "; Actual = " << receivedSequenceId
+                      << "! -> skip" << std::endl;
+        }
+    }))
+    {
+    }
+```
+
+### Server using Listener
+
+At first, the includes for the server port and request-response types and runtime are needed.
+<!-- [geoffrey] [geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp] [iceoryx includes] -->
+```cpp
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/notification_callback.hpp"
+#include "iceoryx_posh/popo/server.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+First, a callback is created that shall be called when the server receives a response.
+In this case the calculation and the sending of the response is done in the listener callback.
+If there are more resource-consuming tasks,
+this could also be outsourced with a thread pool to handle the requests
+
+<!--[geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp][request callback]-->
+```cpp
+void onRequestReceived(iox::popo::Server<AddRequest, AddResponse>* server)
+{
+    while (server->take().and_then([&](const auto& request) {
+        std::cout << APP_NAME << " Got Request: " << request->augend << " + " << request->addend << std::endl;
+
+        server->loan(request)
+            .and_then([&](auto& response) {
+                response->sum = request->augend + request->addend;
+                std::cout << APP_NAME << " Send Response: " << response->sum << std::endl;
+                response.send();
+            })
+            .or_else([](auto& error) {
+                std::cout << "Could not allocate Response! Return value = " << static_cast<uint64_t>(error)
+                          << std::endl;
+            });
+    }))
+    {
+    }
+}
+```
+
+Maybe: The server provides the `take()` method for receiving requests and the `loan()` and `send()` methods
+for sending the responses with the sum of the two numbers.
+
+Next, the iceoryx runtime is initialized.
+<!-- [geoffrey] [iceoryx_examples/request_response/server_cxx_listener.cpp] [initialize runtime] -->
+```cpp
+constexpr char APP_NAME[] = "iox-cpp-request-response-server-basic;
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+```
+
+After creating the runtime, the server port is created based on a ServiceDescription.
+<!--[geoffrey][iceoryx_examples/request_response/server_cxx_listener.cpp][create server]-->
+```cpp
+iox::popo::Server<AddRequest, AddResponse> server({"Example", "Request-Response", "Add"});
```

Comment:
Please also add here a hint that the default options are used and that the user has the ability to change them. Please link therefore the header file to the server options.